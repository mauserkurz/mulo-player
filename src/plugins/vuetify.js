import Vue from 'vue';
import Vuetify from 'vuetify';
import { THEME as theme } from '@/constants';

Vue.use(Vuetify);

export default new Vuetify({ theme });
