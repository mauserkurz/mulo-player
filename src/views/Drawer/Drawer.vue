<template>
  <v-navigation-drawer
    :value="value"
    class="drawer"
    right
    fixed
    temporary
    @input="onInput">
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title class="title">
          <v-btn
            color="primary"
            small
            icon
            @click="onInput(false)">
            <v-icon>mdi-close-circle-outline</v-icon>
          </v-btn>
          Upload music
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>
    <v-divider></v-divider>
    <v-container>
      <v-row
        align="center"
        justify="center">
        <v-col cols="12">
          <slot/>
        </v-col>
      </v-row>
    </v-container>
  </v-navigation-drawer>
</template>

<script>
export default {
  name: 'Drawer',

  props: {
    value: {
      type: Boolean,
      required: true,
    },
  },

  methods: {
    onInput(value) {
      this.$emit('input', value);
    },
  },
};
</script>

<style scoped lang="less">
.drawer {}
</style>
