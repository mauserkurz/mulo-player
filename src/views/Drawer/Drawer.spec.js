import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import vuetify from '@/plugins/vuetify';
import Drawer from './Drawer.vue';

const localVue = createLocalVue();

describe('Component Drawer', () => {
  const defaultProps = { value: true };
  const createWrapper = ({ isWithoutStubs = false, propsData = {}, ...options } = {}) => {
    const renderer = isWithoutStubs ? mount : shallowMount;
    const wrapper = renderer(Drawer, {
      localVue,
      vuetify,
      propsData: Object.assign({}, defaultProps, propsData),
      ...options,
    });

    return { wrapper };
  };

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should emit input event after click on close button', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      wrapper.find('.v-btn').trigger('click');
      expect(wrapper.emitted().input).toEqual([[true], [false]]);
    });
  });
});
