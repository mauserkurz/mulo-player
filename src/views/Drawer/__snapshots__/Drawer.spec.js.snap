// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Drawer snapshots should match snapshot 1`] = `
<aside class="drawer v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--right v-navigation-drawer--temporary theme--light" style="height: 100%; top: 0px; transform: translateX(0%); width: 256px;">
  <div class="v-navigation-drawer__content">
    <div tabindex="-1" class="v-list-item theme--light">
      <div class="v-list-item__content">
        <div class="v-list-item__title title"><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--small primary--text"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close-circle-outline theme--light"></i></span></button>
          Upload music
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="container">
      <div class="row align-center justify-center">
        <div class="col col-12"></div>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</aside>
`;
